minidump-processor/src/process_state.rs

```diff
@@ -416,6 +422,62 @@ impl PossibleBitFlip {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum CrashInconsistency {
+    IntDivByZeroNotPossible,
+    PrivInstructionCrashWithoutPrivInstruction,
+    NonCanonicalAddressFalselyReported,
+    AccessViolationWhenAccessAllowed,
+    CrashingAccessNotFoundInMemoryAccesses,
+}
+
+impl std::fmt::Display for CrashInconsistency {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CrashInconsistency::IntDivByZeroNotPossible => {
+                f.write_str("Crash reason is an integer division by zero but the crashing instruction is not a division")
+            }
+            CrashInconsistency::PrivInstructionCrashWithoutPrivInstruction => {
+                f.write_str("Crash reason is a privileged instruction but crashing instruction is not a privileged one")
+            }
+            CrashInconsistency::NonCanonicalAddressFalselyReported => {
+                f.write_str("Crash address is reported as a non-canonical x86-64 address but the actual address is a canonical one")
+            }
+            CrashInconsistency::AccessViolationWhenAccessAllowed => {
+                f.write_str("Crash reason is access violation exception but access is allowed")
+            }
+            CrashInconsistency::CrashingAccessNotFoundInMemoryAccesses => f.write_str(
+                "Crash address not found among the memory accesses of the crashing instruction",
+            ),
+        }
+    }
+}
+
+impl serde::Serialize for CrashInconsistency {
```

Comment:
Usually people don't directly implement `Serialize` unless necessary. In this case, there are a few alternatives using `#[derive(Serialize)]`:
1. You can use `#[serde(rename_all = "snake_case")]` on the `enum` item, which I believe will result in the same serialization as implemented here.
2. You can use `#[serde(into = "String")]` and add `impl From<CrashInconsistenty> for String` (though this isn't an obvious conversion so this is less preferable).